src/software/thunderscope/replay/proto_player.py

```diff
@@ -130,6 +136,96 @@ def is_log_entry_corrupt(log_entry: bytes, version: int) -> bool:
         except Exception:
             return True
 
+    def build_chunk_index(self, folder_path: os.PathLike) -> dict[str, float]:
+        """Build the chunk index and store the index into the index file
+
+        Note:
+        ----
+        A chunk index entry is a key value pair [filename: start timestamp]
+         - Start timestamp: timestamp of the first log entry in the chunk
+         - Filename: replay chunk file name (%d.reply)
+
+        :param folder_path: folder containing all the replay files
+
+        :return: a dictionary for mapping replay filename to the start
+            timestamp of the first entry in the chunk.
+
+        """
+        chunk_indices: dict[str, float] = dict()
+        for chunk_name in self.sorted_chunks:
+            chunk_data = ProtoPlayer.load_replay_chunk(chunk_name, self.version)
+            if chunk_data:
+                start_timestamp, _, _ = ProtoPlayer.unpack_log_entry(
+                    chunk_data[0], self.version
+                )
+                chunk_indices[os.path.basename(chunk_name)] = start_timestamp
+        if chunk_indices:
+            try:
+                with open(
+                    os.path.join(folder_path, ProtoPlayer.CHUNK_INDEX_FILENAME), "w"
+                ) as index_file:
+                    index_file.write(f"Generated on {time.time()}\n")
```

Comment:
You mentioned that we should store some metadata in the index file header e.g. a version number. I think it would be a good idea to add the version number now so that we don't need some hacky edge case in the future to recognize v1 index files

Reply:
Sure, there is a timestamp of when the index is generated, and a version number of the file in the metadata now.

Reply:
Now the index looks like
```
Version: 1, Generated on 1727935720
1.89261, 0.replay
15.8637, 1.replay
28.327, 2.replay
40.4621, 3.replay
52.8737, 4.replay
64.5744, 5.replay
69.8174, 6.replay
75.1526, 7.replay
85.847, 8.replay
..........., ............
```